src/ui/public/saved_objects/__tests__/saved_objects_client.js

```diff
@@ -0,0 +1,324 @@
+
+
+describe('SavedObjectsClient', () => {
+  const basePath = Math.random().toString(36).substring(7);
+  const sandbox = sinon.sandbox.create();
+
+  let savedObjectsClient;
+
+  beforeEach(() => {
+    savedObjectsClient = new SavedObjectsClient(sinon.stub, basePath);
+    sandbox.stub(savedObjectsClient, '_$http');
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('#_getUrl', () => {
+    it('returns without arguments', () => {
+      const url = savedObjectsClient._getUrl();
+      const expected = `${basePath}/api/saved_objects/`;
+
+      expect(url).to.be(expected);
+    });
+
+    it('appends path', () => {
+      const url = savedObjectsClient._getUrl(['some', 'path']);
+      const expected = `${basePath}/api/saved_objects/some/path`;
+
+      expect(url).to.be(expected);
+    });
+
+    it('appends query', () => {
+      const url = savedObjectsClient._getUrl(['some', 'path'], { foo: 'Foo', bar: 'Bar' });
+      const expected = `${basePath}/api/saved_objects/some/path?foo=Foo&bar=Bar`;
+
+      expect(url).to.be(expected);
+    });
+  });
+
+  describe('#_request', () => {
+    const params = { foo: 'Foo', bar: 'Bar' };
+
+    it('passes options to $http', () => {
+      savedObjectsClient._$http.withArgs({
+        method: 'POST',
+        url: '/api/path',
+        data: params
+      }).returns(Promise.resolve({ data: '' }));
+
+      savedObjectsClient._request('POST', '/api/path', params);
+
+      expect(savedObjectsClient._$http.calledOnce).to.be(true);
+    });
+
+    it('sets params for GET request', () => {
+      savedObjectsClient._$http.withArgs({
+        method: 'GET',
+        url: '/api/path',
+        params: params
+      }).returns(Promise.resolve({ data: '' }));
+
+      savedObjectsClient._request('GET', '/api/path', params);
+
+      expect(savedObjectsClient._$http.calledOnce).to.be(true);
+    });
+
+    it('catches API error', (done) => {
+      const message = 'Request failed';
+
+      savedObjectsClient._$http.returns(Promise.reject({ data: { error: message } }));
+      savedObjectsClient._request('GET', '/api/path', params).then(() => {
+        done('should have thrown');
+      }).catch(e => {
+        expect(e.message).to.eql(message);
+        done();
+      });
+    });
+
+    it('catches API error message', (done) => {
+      const message = 'Request failed';
+
+      savedObjectsClient._$http.returns(Promise.reject({ data: { message: message } }));
+      savedObjectsClient._request('GET', '/api/path', params).then(() => {
+        done('should have thrown');
+      }).catch(e => {
+        expect(e.message).to.eql(message);
+        done();
+      });
+    });
+
+    it('catches API error status', (done) => {
+      savedObjectsClient._$http.returns(Promise.reject({ status: 404 }));
+      savedObjectsClient._request('GET', '/api/path', params).then(() => {
+        done('should have thrown');
+      }).catch(e => {
+        expect(e.message).to.eql('404 Response');
```

Comment:
Thought on [eliminating exposing of the ES query entirely](https://github.com/elastic/kibana/pull/11632/commits/01910ba660c5ad3f34edb973b609b026675f605f)? Logging can be done by enabling `elaticsearch.logQueries` with `--verbose`. Logging things like 404's without verbose or an elevated log level seems a little noisy.